src/ParallelAlgorithms/LinearSolver/Tags.hpp

```diff
@@ -76,10 +77,10 @@ struct OperatorAppliedTo : db::PrefixTag, db::SimpleTag {
  * \brief Holds an `IterationId` that identifies a step in the linear solver
  * algorithm
  */
+template <typename OptionsGroup>
 struct IterationId : db::SimpleTag {
   static std::string name() noexcept {
-    // Add "Linear" prefix to abbreviate the namespace for uniqueness
-    return "LinearIterationId";
+    return option_name<OptionsGroup>() + "(IterationId)";
```

Comment:
This name might be a little tricky, because it is the format typically used for prefix tags, and could be misunderstood in the output to be the tag `OptionGroup<IterationId>`.

Reply:
Would you prefer `IterationId(TheLinearSolver)` over `TheLinearSolver(IterationId)`?

Reply:
I would prefer that. It's still a little ambiguous with prefix tags, but at least it's clear the tag it represents is `IterationId`.